convex/searches.ts

```diff
@@ -43,87 +41,39 @@ export const search = action({
       totalTokens: inputTokens,
       embeddingMs,
     } = await fetchEmbedding(input);
-    const pineconeStart = Date.now();
-    const pinecone = await pineconeIndex();
-    const { matches } = await pinecone.query({
-      queryRequest: {
-        namespace: "chunks",
-        topK,
-        vector: embedding,
-      },
+
+    const result = await ctx.vectorSearch("chunkEmbedding", "by_embedding", {
+      vector: embedding,
+      limit: 16,
     });
-    if (!matches) throw new Error("Pinecone matches are empty");
-    const relatedChunks = matches.map(({ id, score }) => ({
-      id: id as Id<"chunks">,
-      score,
+
+    if (!result) throw new Error("Pinecone matches are empty");
+
+    const relatedChunks = result.map(({ _id, _score }) => ({
+      id: _id as Id<"chunkEmbedding">,
+      score: _score,
     }));
-    const queryMs = Date.now() - pineconeStart;
+
+    console.log({
+      inputTokens,
+      embeddingMs,
+    });
+
     if (searchId) {
-      await upsertVectors(
-        "searches",
-        [{ id: searchId, values: embedding, metadata: { input } }],
-        pinecone
-      );
```

Comment:
We're losing storing the search embedding. I don't think we were using it for anything yet, but one could in theory be searching for previous similar searches and using those results if the search was close enough. It doesn't do much for you, though, since doing the search itself is just as easy.